### PR TITLE
Adding documentation comments

### PR DESCRIPTION
**Pull request by sebastienros**
Adding missing class documentation

### CODE REVIEW

**Comment by Tratcher** [comment]
https://github.com/aspnet/Caching/issues/414

File: src/Microsoft.Extensions.Caching.Abstractions/DistributedCacheEntryOptions.cs [hunk 1]
```diff
@@ -5,6 +5,9 @@
 
 namespace Microsoft.Extensions.Caching.Distributed
 {
+    /// <summary>
+    /// Represents the cache options applied to an entry of the <see cref="IDistributedCache"/> instance.
```
Comment (pranavkm) [inline src/Microsoft.Extensions.Caching.Abstractions/DistributedCacheEntryOptions.cs]: > Provides configuration for an entry in `<see cref="IDistributedCache"/>`

File: src/Microsoft.Extensions.Caching.Abstractions/IDistributedCache.cs [hunk 2]
```diff
@@ -6,22 +6,70 @@
 
 namespace Microsoft.Extensions.Caching.Distributed
 {
+    /// <summary>
+    /// Represents a distributed cache of serialized values.
```
Comment (pranavkm) [inline src/Microsoft.Extensions.Caching.Abstractions/IDistributedCache.cs]: > Contract for a distributed cache of serialized values.

Comment (sebastienros) [reply src/Microsoft.Extensions.Caching.Abstractions/IDistributedCache.cs]: Nope, consistency with `IMemoryCache`

**Pull request merged**